staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -303,6 +303,7 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		return errors.New("can't add TLS certificate callback: transport.Config.TLS.GetCert already set")
 	}
 	c.TLS.GetCert = a.cert
+	c.TLS.CacheTLSTransportOverride = true
```

Comment:
this is not safe... it means clients constructed for distinct client configs can share the same transport, and credentials intended to be provided to one client will be picked up by others instantiated with different configurations in the same process.

Reply:
@liggitt IIUC the difference is only in Client Certificates when considering the Exec plugins and their returned credentials. The token is handled by the RoundTripper so this shouldn't be shared. Could we at least detect when the executable returns empty client certificate and reuse the transport at least in that case?

Reply:
or to retrieve it and include it in the cache key?

Reply:
I suppose we could look into a mechanism for invalidating the cache, but I am not sure if we should be pursuing this

Reply:
@liggitt what is your opinion on this ^?

Reply:
I think trying to make these transports cacheable is fragile and likely to lead to mistakes where credentials are incorrectly shared.

Reply:
ack, let's go with kubectl fix instead